Specify private reserved sigils `^` and `&` (#404)
unicode-org/message-format-wg: spec/syntax.md

#### Reserved

**_Reserved_** annotations start with a reserved character
and are intended for future standardization
as well as private implementation use.

> A reserved sequence MAY be empty or contain arbitrary text.
> A reserved sequence does not include any trailing whitespace.

Reviewer: You are correct to use the term _annotation_ above, but you should be consistent in talking about annotations in the section. This should probably say something like:

```suggestion
A _reserved_ _annotation_ MAY be empty or contain arbitrary text.
This allows maximum flexibility in future standardization. Future definitions 
are expected to define additional semantics and constraints on the contents
of these _annotations_.
Note that a _reserved annotation_ does not include trailing whitespace.
```

Implementations MAY define their own meaning and semantics for
_reserved_ annotations that start with

> the U+0026 AMOERSAND `&` or U+005E CIRCUMFLEX ACCENT `^` characters.

Author: ```suggestion
the U+0026 AMPERSAND `&` or U+005E CIRCUMFLEX ACCENT `^` characters.
```

Implementations MUST NOT define any such meaning or semantics for
_reserved_ annotations that start with any other character;
these are reserved for the use of future versions of this specification.

While a reserved sequence is technically "well-formed",
unrecognized reserved sequences have no meaning and MAY result in errors during formatting.

```abnf

> reserved      = ( future-start / private-start ) reserved-body
> future-start  = "!" / "@" / "#" / "%" / "*" / "<" / ">" / "?" / "~"
> private-start = "^" / "&"

Author: ```suggestion
reserved        = ( reserved-start / private-start ) reserved-body
reserved-start  = "!" / "@" / "#" / "%" / "*" / "<" / ">" / "?" / "~"
private-start   = "^" / "&"
```
